build/bcd-urls.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Loop over, and mutate, all 'browser_compatibility' sections.
+ * BCD data comes froms from a library with `mdn_url`'s that are absolute.
+ * This takes the `mdn_url` and sets it to a URI that can be used when
+ * rendering the BCD table to link to a relative path.
+ *
+ * Also, if enabled, check all of these inner `mdn_url` for flaws.
+ *
+ */
+function normalizeBCDMdnURLs(doc, options) {
+  let checkLinks = false;
+
+  if (options.flawLevels.get("bad_bcd_links") !== FLAW_LEVELS.IGNORE) {
+    checkLinks = true;
+  }
+
+  function addBadBCDLink(query, key, slug, suggestion = null) {
+    if (!("bad_bcd_links" in doc.flaws)) {
+      doc.flaws.bad_bcd_links = [];
+    }
+    doc.flaws.bad_bcd_links.push({
+      slug,
+      suggestion,
+      query,
+      key,
+    });
+  }
+
+  function getPathFromAbsoluteURL(absURL) {
+    // Because the compat data is mutated out of mdn-browser-compat-data,
+    // through our `packageBCD` function, it's very possible that
+    // the `doc[i].type === 'browser_compatibility` has already been
+    // processed.
+    if (!absURL.includes("://")) {
+      return absURL;
+    }
+    const u = url.parse(absURL);
+    if (u.hostname !== "developer.mozilla.org") {
+      // If URL is from a different host, return without modifying it
+      return absURL;
+    }
+    // Return the pathname without docs directory, the base path for the
+    // `Document` component the BCD table is within is /:locale/docs/ so it is
+    // not needed.
+    // The `mdn_url` field in BCD data is always like this:
+    // https://developer.mozilla.org/docs/Web/API/MediaTrackSettings/width
+    // So to get the appropriate slug, in Yari, we have to assume a locale.
+    let slug = u.pathname;
+    if (slug.startsWith("/docs/")) {
+      // Important! For now, to make this a slug we can understand we
+      // have to have a locale and we pick `en-US` as the default.
+      return `/en-US${slug}`;
+    }
+    throw new Error(`not implemented! ${slug}`);
+  }
+
+  doc.body
+    .filter((section) => section.type === "browser_compatibility")
+    .forEach((section) => {
+      Object.entries(section.value.data).forEach(([key, block]) => {
+        // First block from the BCD data does not have its name as the root key
+        // so mdn_url is accessible at the root. If the block has a key for
+        // `__compat` it is not the first block, and the information is nested
+        // under `__compat`.
+        if (block.__compat) {
+          block = block.__compat;
+        }
+        if (block.mdn_url) {
+          const url = getPathFromAbsoluteURL(block.mdn_url);
+          block.mdn_url = url;
+          if (checkLinks && url.startsWith("/")) {
+            // Now we need to scrutinize if that's a url we can fully
+            // recognize. (But only if it's a relative link)
+            const urlLC = url.toLowerCase();
+            const found = Document.findByURL(urlLC, {
+              metadata: true,
+            });
```

Comment:
Just thinking out loud. Since we're just using this as a boolean test to see if the URL exists, we don't care about the data returned, but it's probably better to stick with what you have since it has a better chance of already being memoized, instead of doing something like:
```js
            const found = Document.findByURL(urlLC, {
              body: false,
              metadata: false,
            });
```

Reply:
You're thinking something like a `headByURL` as form of getting a yes or no that it can resolve. 
Either way, the memoization serves has very well here because either this is the call that warms it up for when those pages are rendered, or it's already warmed up.